From: Ilya Dryomov <idryomov@gmail.com>
Date: Fri, 14 Jul 2017 16:08:54 +0200
Subject: libceph: don't call encode_request_finish() on MOSDBackoff messages
Git-commit: 914902af4f271884968f72c4fa144c723be2a699
Patch-mainline: v4.13-rc2
References: bsc#1048228

encode_request_finish() is for MOSDOp messages.  Calling it on
MOSDBackoff ack-block messages corrupts them.

Fixes: a02a946dfe96 ("libceph: respect RADOS_BACKOFF backoffs")
Signed-off-by: Ilya Dryomov <idryomov@gmail.com>
Acked-by: Luis Henriques <lhenriques@suse.com>
---
 net/ceph/osd_client.c |    5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

--- a/net/ceph/osd_client.c
+++ b/net/ceph/osd_client.c
@@ -5477,7 +5477,10 @@ static int invalidate_authorizer(struct
 
 static void osd_reencode_message(struct ceph_msg *msg)
 {
-	encode_request_finish(msg);
+	int type = le16_to_cpu(msg->hdr.type);
+
+	if (type == CEPH_MSG_OSD_OP)
+		encode_request_finish(msg);
 }
 
 static int osd_sign_message(struct ceph_msg *msg)
